examples/step-67/step-67.cc

```diff
@@ -0,0 +1,2305 @@
+    const std::pair<unsigned int, unsigned int> &     face_range) const
+  {
+    FEFaceEvaluation<dim, degree, n_points_1d, dim + 2, Number> phi(data, true);
+
+    for (unsigned int face = face_range.first; face < face_range.second; face++)
+      {
+        phi.reinit(face);
+        phi.gather_evaluate(src, true, false);
+
+        for (unsigned int q = 0; q < phi.n_q_points; ++q)
+          {
+            const auto w_m    = phi.get_value(q);
+            const auto normal = phi.get_normal_vector(q);
+
+            auto rho_u_dot_n = w_m[1] * normal[0];
+            for (unsigned int d = 1; d < dim; ++d)
+              rho_u_dot_n += w_m[1 + d] * normal[d];
+
+            bool at_outflow = false;
+
+            Tensor<1, dim + 2, VectorizedArray<Number>> w_p;
+            const auto boundary_id = data.get_boundary_id(face);
+            if (wall_boundaries.find(boundary_id) != wall_boundaries.end())
+              {
+                w_p[0] = w_m[0];
+                for (unsigned int d = 0; d < dim; ++d)
+                  w_p[d + 1] = w_m[d + 1] - 2. * rho_u_dot_n * normal[d];
+                w_p[dim + 1] = w_m[dim + 1];
+              }
+            else if (inflow_boundaries.find(boundary_id) !=
+                     inflow_boundaries.end())
+              w_p =
+                evaluate_function(*inflow_boundaries.find(boundary_id)->second,
+                                  phi.quadrature_point(q));
+            else if (subsonic_outflow_boundaries.find(boundary_id) !=
+                     subsonic_outflow_boundaries.end())
+              {
+                w_p          = w_m;
+                w_p[dim + 1] = evaluate_function(
+                  *subsonic_outflow_boundaries.find(boundary_id)->second,
+                  phi.quadrature_point(q),
+                  dim + 1);
+                at_outflow = true;
+              }
+            else if (supersonic_outflow_boundaries.find(boundary_id) !=
+                     supersonic_outflow_boundaries.end())
+              {
+                w_p        = w_m;
+                at_outflow = true;
+              }
+            else
+              AssertThrow(false,
+                          ExcMessage("Unknown boundary id, did "
+                                     "you set a boundary condition?"));
+
+            auto flux = euler_numerical_flux<dim>(w_m, w_p, normal);
+
+            if (at_outflow)
+              {
+                for (unsigned int v = 0;
+                     v < VectorizedArray<Number>::n_array_elements;
+                     ++v)
+                  {
+                    if (rho_u_dot_n[v] < -1e-12)
+                      for (unsigned int d = 0; d < dim; ++d)
+                        flux[d + 1][v] = 0.;
+                  }
+              }
+
+            phi.submit_value(-flux, q);
+          }
+
+        phi.integrate_scatter(true, false, dst);
+      }
+  }
+
+
+
+  // This function implements the inverse mass matr
```

Comment:
This is not updated yet, is it?

Reply:
I'm waiting for PR #9231 to be reviewed and merged, then I will add that code.